src/expandableCalendar/index.tsx

```diff
@@ -255,29 +259,35 @@ const ExpandableCalendar = (props: ExpandableCalendarProps) => {
     }
   }, []);
 
-  useEffect(() => {
-    // date was changed from AgendaList, arrows or scroll
-    scrollToDate(date);
-  }, [date]);
-
   const handleScreenReaderStatus = (screenReaderEnabled: any) => {
     setScreenReaderEnabled(screenReaderEnabled);
   };
 
   /** Scroll */
+  const autoScroll = useRef(false);
+
+  const scrollToDate = throttle((date: string) => {
+    autoScroll.current = true;
 
-  const scrollToDate = (date: string) => {
     if (!horizontal) {
       calendarList?.current?.scrollToDay(date, 0, true);
     } else if (getYear(date) !== visibleYear.current || getMonth(date) !== visibleMonth.current) {
       // don't scroll if the month is already visible
       calendarList?.current?.scrollToMonth(date);
     }
-  };
+  }, 100, {leading: false, trailing: true});
+
+  /** Update Date */
 
-  const scrollPage = useCallback((next: boolean) => {
+  const newTempDate = useRef(date);
+
+  const _setDate = debounce((updateSource = updateSources.PAGE_SCROLL) => {
```

Comment:
Same here, the debounce method should be wrapped with useCallback

Reply:
Done